### PR TITLE
tests(visual): add Argos CI

### PR DESCRIPTION
**Pull request by oliviertassinari**
This PR is a continuation of https://github.com/algolia/instantsearch.js/pull/2166.
We are integrating [Storybook](https://github.com/storybooks/storybook), [Happo](https://github.com/Galooshi/happo) and [Argos-CI](https://www.argos-ci.com/).

- Storybook responsibility is to aggregate the React Components through stories.
- Happo responsibility is to generate screenshots based on Storybook's stories.
- Argos-CI responsibility is to handle the diffs and dev workflow based on Happo's screenshots.

The following video is an example of what Happo is doing for us:
juin-18-2017 00-31-58

Things we need to get right:
- [x] Better name handling
- [x] Test stability
- [x] Get https://github.com/Galooshi/happo/pull/209 released
- [x] Update the normalize.js script for a more up to date version.

*For more context on [the available solutions](https://storybook.js.org/testing/react-ui-testing/)*

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,9 +70,13 @@
     "eslint-plugin-jasmine": "^2.1.0",
     "eslint-plugin-jest": "^19.0.1",
     "eslint-plugin-react": "^6.10.3",
+    "fs-extra": "^3.0.1",
     "gh-pages": "^0.12.0",
     "glob": "^7.1.1",
     "google-map-react": "^0.22.3",
+    "happo-olivier": "^5.0.2",
```
Comment (oliviertassinari) [inline package.json]: custom fork waiting for the fix to be released.

**Comment by mthuret** [comment]
@oliviertassinari that's super nice! Love it!

What do you mean with the task "Test stability"?

**Comment by oliviertassinari** [comment]
>  that's super nice! Love it!

I'm happy to hear it :).

> What do you mean with the task "Test stability"?

I wanted to make sure the pending http request trick was working with all the stories. That seems to be fine.

**Comment by mthuret** [comment]
Let's wait for that fix being released then :)

File: test/regressions/XMLHttpRequest.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+(function() {
```
Comment (vvo) [inline test/regressions/XMLHttpRequest.js]: Can you add a comment to inform what is this file and why we need it?
Then link to this PR.

File: test/regressions/normalize.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const path = require('path');
```
Comment (vvo) [inline test/regressions/normalize.js]: Can you add a comment on what is this file doing?

**Comment by vvo** [comment]
This looks cool!

What would we have to do to get those screenshots/diffs to be ran in multiple browsers? Would having some functional tests going through all stories help?

File: test/regressions/normalize.js (hunk 3 above)
Comment (vvo) [inline test/regressions/normalize.js]: Needs eslint comment

**Comment by oliviertassinari** [comment]
> What would we have to do to get those screenshots/diffs to be ran in multiple browsers? Would having some functional tests going through all stories help?

@vvo Happo doesn't support multiple browsers yet. We would either have to write a [chrome-headless](https://github.com/argos-ci/argos/tree/master/examples/with-chrome-headless) target plugin or a selenium target plugin for even more browser coverage like we could go through Browserstack or Saucelab.
On Material-UI we are using [vrtest](https://github.com/argos-ci/argos/tree/master/examples/with-vrtest) with Chrome and Selenium in a docker, making cross-browser testing quite easy.

File: test/regressions/XMLHttpRequest.js (hunk 2 above)
Comment (oliviertassinari) [inline test/regressions/XMLHttpRequest.js]: Sure

File: test/regressions/normalize.js (hunk 3 above)
Comment (oliviertassinari) [inline test/regressions/normalize.js]: Sure

**Comment by vvo** [comment]
As for screenshots and diffing I would say a ultimate solution would be to have a selenium pluggin that is easy to use yes so that we can take screenshots from many browsers at the same time. Ultimately that's what we would like.

**Comment by mthuret** [comment]
@oliviertassinari if you wish to be compatible with storybook directly you could create an addon for it. The user will have the ability to use you as a decorator for all their stories which gave you the opportunity to do whatever you want for each on of theme. 

Check this documentation: https://storybook.js.org/addons/introduction/

**Comment by oliviertassinari** [comment]
@mthuret I didn't know about storybook plugins, thanks for sharing it. After going throw the documentation, I don't see how that could help us with the screenshot generation. All we need are the components and the styles when not using a CSS-in-JSS solution.

**Comment by oliviertassinari** [comment]
> Ultimately that's what we would like.

@vvo Shouldn't be too hard to write given we have [the source code of vrtest](https://github.com/nathanmarks/vrtest) doing that, I'm wondering if Happo has any plan for that.

File: docgen/assets/js/sidebar.js [hunk 4]
```diff
@@ -32,7 +32,8 @@ function sidebarFollowScroll(sidebarContainer) {
         sidebarContainer.style.top = null;
       }
       sidebarContainer.classList.add('fixed');
-      linksContainer.style.maxHeight = `calc(100vh - ${titleHeight + navHeight}px)`;
+      linksContainer.style.maxHeight = `calc(100vh - ${titleHeight +
```
Comment (oliviertassinari) [inline docgen/assets/js/sidebar.js]: I have been running `yarn reformat`.

Comment (oliviertassinari) [inline docgen/assets/js/sidebar.js]: Feel free to do so on master, I will rebase if you need a less noising PR.

**Comment by vvo** [comment]
LGTM

**Comment by oliviertassinari** [comment]
@mthuret Let me know if you experience any issue why that solution. I'm here to help.

**Pull request merged**